pkg/cluster/delete_test.go

```diff
@@ -0,0 +1,91 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	mgmtfeatures "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2019-07-01/features"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/mock/gomock"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	mock_network "github.com/Azure/ARO-RP/pkg/util/mocks/azureclient/mgmt/network"
+	mock_env "github.com/Azure/ARO-RP/pkg/util/mocks/env"
+)
+
+func getNic(subscription, resourceGroup, nicName, provisioningState string) mgmtfeatures.GenericResourceExpanded {
+	return mgmtfeatures.GenericResourceExpanded{
+		Name:              to.StringPtr(nicName),
+		ID:                to.StringPtr(fmt.Sprintf("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/networkInterfaces/%s", subscription, resourceGroup, nicName)),
+		ProvisioningState: to.StringPtr(provisioningState),
+	}
+}
+
+func TestDeleteNic(t *testing.T) {
+	ctx := context.Background()
+	subscription := "00000000-0000-0000-0000-000000000000"
+	clusterRG := "cluster-rg"
+	nicName := "nic-name"
+	location := "eastus"
+
+	tests := []struct {
+		name              string
+		mocks             func(*mock_network.MockInterfacesClient)
+		provisioningState string
+	}{
```

Comment:
Can we add two more test cases for when `CreateOrUpdateAndWait` returns an error as well as when `DeleteAndWait` returns an error? We can add a `wantErr` to this test struct and check for it on L86. [Here is an example](https://github.com/Azure/ARO-RP/blob/acee23ae62df6172916fa58a697b99d6d33d71f2/pkg/util/instancemetadata/prod_test.go#L141)

Reply:
@s-amann I added test cases you asked for, I rebased and force pushed the branch again so please see new commits and let me know if you have any suggestions!